source/adapters/level_zero/usm.cpp

```diff
@@ -295,16 +295,23 @@ UR_APIEXPORT ur_result_t UR_APICALL urUSMHostAlloc(
   // find the allocator depending on context as we do for Shared and Device
   // allocations.
   umf_memory_pool_handle_t hPoolInternal = nullptr;
+  usm::pool_descriptor Desc = {nullptr, Context, nullptr, UR_USM_TYPE_HOST,
+                               false};
   if (!UseUSMAllocator ||
       // L0 spec says that allocation fails if Alignment != 2^n, in order to
       // keep the same behavior for the allocator, just call L0 API directly and
       // return the error code.
       ((Align & (Align - 1)) != 0)) {
-    hPoolInternal = Context->HostMemProxyPool.get();
+    auto hPoolInternalOpt = Context->ProxyPoolManager.getPool(Desc);
+    hPoolInternal = hPoolInternalOpt.value();
```

Comment:
These calls throw an exception if std::optional does not contain a value. Add a check:
```
if (hPoolInternalOpt) {
    ...
}
```